request_test.go

```diff
@@ -607,6 +607,164 @@ func TestUnmarshalRelationships(t *testing.T) {
 	}
 }
 
+type Image struct {
+	ID  int    `jsonapi:"primary,images"`
+	Src string `jsonapi:"attr,src"`
+}
+
+type Video struct {
+	ID       int    `jsonapi:"primary,videos"`
+	Captions string `jsonapi:"attr,captions"`
+}
+
+type OneOfMedia struct {
+	Image *Image
+	Video *Video
+}
+
+func Test_UnmarshalPayload_polymorphicRelations(t *testing.T) {
+	type pointerToOne struct {
+		ID    int           `jsonapi:"primary,blogs"`
+		Title string        `jsonapi:"attr,title"`
+		Hero  *OneOfMedia   `jsonapi:"polyrelation,hero-media,omitempty"`
+		Media []*OneOfMedia `jsonapi:"polyrelation,media,omitempty"`
+	}
+
+	in := bytes.NewReader([]byte(`{
+		"data": {
+			"type": "blogs",
+			"id":   "3",
+			"attributes": {
+				"title": "Hello, World"
+			},
+			"relationships": {
+				"hero-media": {
+					"data": {
+						"type": "videos",
+						"id":   "1",
+						"attributes": {
```

Comment:
I dont think that you would actually get attributed populated like this in jsonapi responses. The relationships object should be a single [resource linkage](https://jsonapi.org/format/#document-resource-object-linkage) (or an array of them) with just a `type` and `id`.

The library does some [magical sideloading](https://github.com/hashicorp/jsonapi/blob/main/request.go#L382) of data included in the `included` section to populate the attributes of the related modes.

For something like this, I would expect the response to look like:
```
{
"relationships": {
  "hero-media": {
    "data": {
      "type": "videos",
      "id":   "1"
    }
  }
},
"included": [
  {
    "type": "videos",
    "id": "1",
    "attributes": {
      "captions": "It's Awesome!"
    }
  }
]
}
```

Reply:
Yes, you are right. That would be a much better way to test this. I corrected the fixture ✅